### ISSUE TITLE
Page layout

### ISSUE THREAD

**Issue by feloy** [containers/podman-desktop#7757]
### Is your enhancement related to a problem? Please describe

We have several page layouts (NavPage, DetailsPage, FormPage, SettingsPage). All are very similar in their design, with a few specificities for each:

- a search bar
- a breadcrumb (and a close button)
- an icon in front of the title
- a specific set of colors (background, text, card background, card text, ...)

### Describe the solution you'd like

We could consider having only one page layout, with all these specificities as parameters.
That would help factorise the design of all the pages, and have more consistent pages.

For specific colors, we could consider passing these specific colors in the context, so the colors used in the page layout and inside the fragments would use the local colors instead of colors from the registry.

Schematically:
```
<Page let:bg="pd-content-bg" let:text="pd-content-text">
  <div class="bg-${bg} text-${text}">Some text using colors from the context</div>
</Page>
```

**Comment by ekidneyrh** [comment]
@feloy  I briefly looked around the app and all the pages follow the same structure just +/- some elements

- Breadcrumb
- Icon + Header + buttons
- Subtitle (can be text or link)
- Search bar
- Tabs
- Content (table, list)

I probably shouldn't have assigned this to myself, as I can't do more in terms of the code. Would it be beneficial to have me make a diagram for this? Maybe include the colours on it too? WDYT

**Comment by deboer-tim** [comment]
I know there are some differences between the main nav page and form page header layout, it would probably help to understand how the uber page should be laid out in terms of flex box, e.g. if there are cases where we have no breadcrumb but an X how the header changes. The Container Details page is one of the more complicated ones where I'm not sure the layout is great at different sizes.

But yes, otherwise I think we should assign for implementation.

**Comment by feloy** [comment]
> @feloy I briefly looked around the app and all the pages follow the same structure just +/- some elements
> 
> * Breadcrumb
> * Icon + Header + buttons
> * Subtitle (can be text or link)
> * Search bar
> * Tabs
> * Content (table, list)
> 
> I probably shouldn't have assigned this to myself, as I can't do more in terms of the code. Would it be beneficial to have me make a diagram for this? Maybe include the colours on it too? WDYT

@ekidneyrh, yes some mockups would be helpful IMO, representing the page with the different options set or not (the page with or without breadcrumb, the page with or without icon, etc). And also which colors would be different based on the context (content / inverted content / details / ...)

**Comment by ekidneyrh** [comment]
> @ekidneyrh, yes some mockups would be helpful IMO, representing the page with the different options set or not (the page with or without breadcrumb, the page with or without icon, etc). And also which colors would be different based on the context (content / inverted content / details / ...)

Okay great I'll get working on it :)

**Comment by ekidneyrh** [comment]
I started on the header section of each page first, as there's a lot there. 

For the colours, I have mapped out all the different components that can be present on a page's header:

[Header_layout.pdf](https://github.com/user-attachments/files/16335665/Header_layout.pdf)
![Header_layout](https://github.com/user-attachments/assets/487260d1-8cec-4cb5-b6e0-d9739fff06ff)

Here's all the variations of the header throughout the app. Every component should stay the same size no matter what variation is used. We can also discuss at some point finding ways of reducing the amount of variation.

![image](https://github.com/user-attachments/assets/1fb4701c-c461-45af-9525-a1c576bdc29d)

**Comment by ekidneyrh** [comment]
Here's all the variations of the body content throughout the app. I think we should find a way to make pages using cards look more related? There's a lot of variation atm. 

![Screenshot from 2024-07-25 12-51-28](https://github.com/user-attachments/assets/703e5bb9-51b1-4ade-be60-0c4296a3ec3d)

**Comment by ekidneyrh** [comment]
I'm currently working on a document for this to explain each of the components on a page and the different types of headers and bodies a page can have. https://design.penpot.app/#/view/f336a94c-01dc-8157-8004-994adf55cce2?page-id=d328376a-7f67-80b6-8004-bc04b03d7129&section=interactions&index=0&share-id=5d3c93f2-4b22-803f-8004-d84a144a2aff

**Comment by ekidneyrh** [comment]
Had a discussion with @MariaLeonova and was wondering if this is something that will be recreated on our storybook? Is this the correct link for it? https://podman-desktop.io/storybook 

I see this ticket https://github.com/containers/podman-desktop/issues/7441 for adding stories. 

Do we think making a penpot file for it to be recreated in storybook is redundant?

**Comment by slemeur** [comment]
It feels to me that this ticket is in stale status. 
I don't think we should have that in a penpot file, but rather in the storybook. 

@feloy: Do you have everything needed?

**Comment by ekidneyrh** [comment]
Closing as stale per above comment
